cpp/src/sort/faster_sort_column_impl.cuh

```diff
@@ -111,12 +122,45 @@ struct faster_sorted_order_fn {
 
   template <typename T>
   void operator()(mutable_column_view& input,
-                  mutable_column_view& indices,
+                  mutable_column_view& indices,  // no longer preload this
                   bool ascending,
                   rmm::cuda_stream_view stream)
-    requires(is_supported<T>() and !cudf::is_chrono<T>())
+    requires(cudf::is_floating_point<T>())
   {
-    faster_sort<T>(input, indices, ascending, stream);
+    auto pair_in  = rmm::device_uvector<float_pair<T>>(input.size(), stream);
+    auto d_in     = pair_in.begin();
+    auto pair_out = rmm::device_uvector<float_pair<T>>(input.size(), stream);
+    auto d_out    = pair_out.begin();
```

Comment:
Is `d_out` used? Or is there something we need to do to update `mutable_column_view input`? It seems like we only care about giving back the sorting indices (`auto dv_out = indices.begin<cudf::size_type>()`), and the sorted input itself is thrown away, but I may be misreading this.

Reply:
(I'm wondering if we are currently doing a gather that we could avoid by reusing the sorted output column for single-column sorts.)

Reply:
The `d_out` is not used but necessary to fulfill the API. 
This PR is only used for `sorted_order` . I should make that clear in the description.
I will do a follow-on PR for just plain sort that will be similar.

Reply:
Can we fulfill the API with a discard iterator and avoid allocating this memory?

Reply:
Nope. The API does not accept iterators. This is what `thrust::sort_by_key()` is doing internally so there is really no new memory allocations required here other the `int` part of the decomposer's `float_pair`.

Reply:
Great. Then this is fine as it is.